### PR TITLE
Advanced solution for es6/mutate-an-array-declared-with-const

### PR DESCRIPTION
**Pull request by imambungo**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

**Review by ezioda004 — COMMENTED** [review]
Hi @imambungo,

Could you explain the reason for updating the curriculum file? The changes you've made solves a non-existing problem, unless I'm missing something.

**Comment by imambungo** [comment]
Hi @ezioda004,

Sorry, what i mean by problem is the challenge of [the lesson](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const/), which is to change an array from [5, 7, 2] to [2, 5, 7].

From my point of view, the challenge is to move the last element of the array to the front. Maybe that's why the challenge creator ask user to change the array to [2, 5, 7], not to some random array like [3, 9, 0].

The basic solution of this problem is to assign each element with the designated value which is so inefficient if the array has a lot of element. It is way better to just `pop()` the last element and `unshift()` it to front.

If there are only 3 element needed to change, i think people will lazy to think and use the right solution the challenge creator asked. Even worse, they don't realize it at all. That's why i increase the number of element in the array by 1, to wake people up of the real intention of the challenge creator. Maybe many people will realize it if there are 4 elements in the array.

Even if people still just use the basic solution (assign each element manually), they will pass the challenge anyway. I'm just trying to tell people to solve the problem efficiently.

Sorry for bad english, i'm still learning.

**Comment by ezioda004** [comment]
@imambungo The challenge is called 
>ES6: **Mutate** an Array Declared with const

So even if the user uses `.pop()` or any element, it'll be effectively *mutating* the array. The objective of the challenge is to learn that even with `const` declaration, object/arrays can be mutated.

**Comment by imambungo** [comment]
@ezioda004 Okay, so if what i said the lesson is the challenge, then what i said the challenge must be the tests?

By the way, i have just test it locally, the tests is work as expected. The hint/guide page also works fine.

mypr

theguide

**Comment by ezioda004** [comment]
@imambungo Sorry, I dont get what you said. The reasons you've mentioned for the change are not compelling enough.
I'd like to hear others opinion on this change.

**Comment by lynxlynxlynx** [comment]
While the extra element may not add much, I see no problem with the extra, scalable solution and the rest of the fixes.

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const/index.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Reassign the values of the `const` variable `s` using various element assignment
 
 > _try to solve the problem now_
 
-## ![:speech_balloon:](https://forum.freecodecamp.com/images/emoji/emoji_one/speech_balloon.png?v=3 ":speech_balloon:") Hint: 1
+## ![:speech_balloon:](https://forum.freecodecamp.com/images/emoji/emoji_one/speech_balloon.png?v=3 ":speech_balloon:") Hint: 2
 
 *   To access array value use array[index]
```
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const/index.md]:
```suggestion
*   To access array value use `array[index]`
```
Please put this in code format

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Check my comment

File: guide/english/certifications/javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const/index.md [hunk 2]
```diff
@@ -44,6 +45,21 @@ Reassign the values of the `const` variable `s` using various element assignment
...
+## ![:beginner:](https://forum.freecodecamp.com/images/emoji/emoji_one/beginner.png?v=3 ":beginner:") Advanced Code Solution:
+```javascript
+    const s = [5, 7, 2, 8];
+    function editInPlace() {
+      "use strict";
+      s.unshift(s.pop());
+    }
+    editInPlace();
+```
+![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://codepen.io/imambungo/pen/rQjXpX' target='_blank' rel='nofollow'>Run Code</a>
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const/index.md]: Please remove this line as we no longer want external links like this to personal accounts.
```suggestion
```

**Review by RandellDawson — DISMISSED** [review]
See my request for changes.

**Comment by RandellDawson** [comment]
I have added my request for changes above in case the consensus ends up being to merge this change.  However, I do not think it is a necessary change at all, since the point of the challenge is not about making the most efficient solution here, but instead is to show an array declared with `const` is still immutable. 

@imambungo Who's to say that ` [2, 5, 7]` is not random?

I say we close this PR.

**Comment by thecodingaviator** [comment]
@imambungo thank you for opening this pull request. For the same reason as mentioned by @RandellDawson above I'm closing this PR.

This is a message notifying you that we’ve reviewed your pull request and have decided not to merge it. We would welcome future pull requests from you.

Thank you and happy coding. ✈️

**Pull request closed without merging**